packages/neon_framework/test/request_manager_test.dart

```diff
@@ -0,0 +1,927 @@
+// ignore_for_file: unnecessary_lambdas
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:neon_framework/src/bloc/result.dart';
+import 'package:neon_framework/src/utils/request_manager.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:rxdart/rxdart.dart';
+
+class MockCallbackFunction<T> extends Mock {
+  FutureOr<T> call();
+}
+
+class MockedCache extends Mock implements Cache {}
+
+void main() {
+  tearDown(() {
+    RequestManager.instance = null;
+    Cache.instance = null;
+  });
+
+  group('Cache', () {
+    test('singleton', () {
+      expect(identical(Cache(), Cache()), isTrue);
+    });
+  });
```

Comment:
Cache should also have tests

Reply:
https://github.com/nextcloud/neon/pull/1466#issuecomment-1895546285

If we add tests it should be for the lowest supported sqlite version.
I'll open an issue to investigate this. Maybe we can specify a custom binary to use and add it to our test setup.